py/mpconfig: Enable 2-argument built-in next() at basic feature level.

dpgeorge opened the pull request:
### Summary

Enable the 2-argument form of the built-in `next()` function at the basic feature level (which is one above core features).

This is a pretty fundamental built-in and having CPython-compatible behaviour is beneficial.  The code size increase is not much, and ports/boards can still disable it if needed to save space.

Addresses issue #5384.

### Testing

CI should test this.

### Trade-offs and Alternatives

Could leave it disabled and document better the differences.  But I think it's worth the increase in code size, most big boards have enough space.

dpgeorge committed:
py/mpconfig: Enable 2-argument built-in next() at basic feature level.

This is a pretty fundamental built-in and having CPython-compatible
behaviour is beneficial.  The code size increase is not much, and
ports/boards can still disable it if needed to save space.

Addresses issue #5384.

Signed-off-by: Damien George <damien@micropython.org>

projectgus commented:
Change LGTM but we should probably update or remove `tests/cpydiff/builtin_next_arg2.py` at the same time. (I'm not sure what the approach is for CPython differences that only exist at certain feature levels?)

dpgeorge committed:
tests/cpydiff: Remove builtin_next_arg2.py difference.

Because 2-arg `next()` is implemented, and now enabled at the basic feature
level.

Signed-off-by: Damien George <damien@micropython.org>
